### ISSUE TITLE
Codemirror state not being properly updated with latest scratch contents after saving and then navigating to and from the scratch page

### ISSUE THREAD

**Issue by ethteck** [decompme/decomp.me#711]
https://discord.com/channels/897066363951128586/897066363951128590/1083756322874458253

Video that reproduces it above.

I can confirm this happens both on local builds and on local `yarn dev`. I also noticed that the network request for the scratch data includes the correct, up-to-date source code, and the site even knows that what is in the codemirror box is out of date because it says "unsaved" as soon as you navigate to the page. So as I thought,
<img width="296" alt="image" src="https://user-images.githubusercontent.com/2985314/227715336-faac338c-afc3-4f61-b288-9028154c6722.png">

**Comment by ethteck** [comment]
https://user-images.githubusercontent.com/2985314/227766622-566a8693-90aa-4fc5-a54b-35cddbb68fdd.mp4

Putting here what I wrote in discord:

this PR fixes the bug, but there's a slightly annoying delay after returning to the page before the contents update. it looks like the delay corresponds to how long it takes to run `GET` on the scratch again after loading the page. what I'm wondering is why we can't use the most recent result from api.get(scratch.url) instead. we have a fresh result that's from before we went to the profile page, already.

 Can we instead update `initialScratch` every time we do a GET on the scratch page? Or idk, in general something feels like this shouldn't be as annoying as it is to work around
